Tests/Unit/UriGeneratorTest.php

```diff
@@ -76,16 +76,39 @@ public function provideGenerateUri()
                     ),
                 ),
             ),
+            array(
+                '/this/is/{unknown_token}/life',
+                null,
+                array(),
+                array('InvalidArgumentException', 'Unknown token "unknown_token"'),
+            ),
+            array(
+                'this/is/not/absolute',
```

Comment:
It is very important that the URI is absolute -- I wonder if we should check if the URI is valid also, but that might be a bigger performance issue.. but then maybe we should be caching this.